### PR TITLE
Support EmbedderExternalTextureGL for impeller

### PR DESCRIPTION
**Pull request by xiaowei-guan**
Support EmbedderExternalTextureGL for impeller, the EvasGL part has not been impelemented yet.
#15

### CODE REVIEW

File: flutter/shell/platform/tizen/external_texture_surface_egl_impeller.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
+// Copyright 2020 Samsung Electronics Co., Ltd. All rights reserved.
```
[the remark this answers is not in the document]
Comment (xiaowei-guan) [reply flutter/shell/platform/tizen/external_texture_surface_egl_impeller.h]: Done

File: flutter/shell/platform/tizen/external_texture_pixel_egl_impeller.cc [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  width = pixel_buffer->width;
+  height = pixel_buffer->height;
+
+  // Populate the texture object used by the engine.
+  opengl_texture->impeller_texture_type =
+      FlutterGLImpellerTextureType::kFlutterGLImpellerTexturePixelBuffer;
+  opengl_texture->buffer = pixel_buffer->buffer;
+  opengl_texture->buffer_size =
+      size_t(pixel_buffer->width) * size_t(pixel_buffer->height) * (32 >> 3);
```
[the remark this answers is not in the document]
Comment (xiaowei-guan) [reply flutter/shell/platform/tizen/external_texture_pixel_egl_impeller.cc]: Fixed

File: flutter/shell/platform/tizen/external_texture_surface_egl_impeller.h [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  bool OnBind();
+  FlutterDesktopGpuSurfaceTextureCallback texture_callback_ = nullptr;
+  void* user_data_ = nullptr;
+  EGLImageKHR egl_src_image_ = nullptr;
+  void* last_surface_handle_ = nullptr;
+};
+
+}  // namespace flutter
+
+#endif  // EMBEDDER_EXTERNAL_TEXTURE_SURFACE_EGL_H_
```
[the remark this answers is not in the document]
Comment (xiaowei-guan) [reply flutter/shell/platform/tizen/external_texture_surface_egl_impeller.h]: Fixed

File: flutter/shell/platform/tizen/external_texture_surface_egl_impeller.cc [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+
+bool ExternalTextureSurfaceEGLImpeller::OnBindCallback(void* user_data) {
+  ExternalTextureSurfaceEGLImpeller* self =
+      static_cast<ExternalTextureSurfaceEGLImpeller*>(user_data);
+  return self->OnBind();
+}
+
+bool ExternalTextureSurfaceEGLImpeller::OnBind() {
+  if (!egl_src_image_)
+    return false;
```
[the remark this answers is not in the document]
Comment (xiaowei-guan) [reply flutter/shell/platform/tizen/external_texture_surface_egl_impeller.cc]: Fixed

**Review by JSUYA — APPROVED** [review]

File: flutter/shell/platform/tizen/external_texture_pixel_egl_impeller.cc [hunk 5]
```diff
@@ -0,0 +1,53 @@
+// Copyright 2020 Samsung Electronics Co., Ltd. All rights reserved.
```
[the remark this answers is not in the document]
Comment (xiaowei-guan) [reply flutter/shell/platform/tizen/external_texture_pixel_egl_impeller.cc]: Fixed

File: flutter/shell/platform/tizen/external_texture_surface_egl_impeller.cc [hunk 6]
```diff
@@ -0,0 +1,189 @@
+// Copyright 2020 Samsung Electronics Co., Ltd. All rights reserved.
```
[the remark this answers is not in the document]
Comment (xiaowei-guan) [reply flutter/shell/platform/tizen/external_texture_surface_egl_impeller.cc]: Fixed

**Pull request merged**